cmd/sops/subcommand/exec/exec.go

```diff
@@ -0,0 +1,100 @@
+package exec
+
+import (
+	"log"
+	"bytes"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"runtime"
+	"strings"
+)
+
+type ExecOpts struct {
+	Command string
+	Plaintext []byte
+	Background bool
+	Fifo bool
+	User string
+}
+
+func GetFile(dir string) *os.File {
+	handle, err := ioutil.TempFile(dir, "tmp-file")
+	if err != nil {
+		log.Fatal(err)
+	}
+	return handle
+}
+
+func ExecWithFile(opts ExecOpts) {
+	if opts.User != "" {
+		SwitchUser(opts.User)
+	}
+
+	if runtime.GOOS == "windows" && opts.Fifo {
+		log.Print("no fifos on windows, use --no-fifo next time")
+		opts.Fifo = false
+	}
+
+	dir, err := ioutil.TempDir("/tmp/", ".sops")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer os.RemoveAll(dir)
+
+	var filename string
+	if opts.Fifo {
+		// fifo handling needs to be async, even opening to write
+		// will block if there is no reader present
+		filename = GetPipe(dir)
+		go WritePipe(filename, opts.Plaintext)
+	} else {
+		handle := GetFile(dir)
+		handle.Write(opts.Plaintext)
+		handle.Close()
+		filename = handle.Name()
+	}
+
+	placeholdered := strings.Replace(opts.Command, "{}", filename, -1)
+	cmd := exec.Command("/bin/sh", "-c", placeholdered)
```

Comment:
Why not call the command directly ? calling through `/bin/sh` will break in native Windows.

Reply:
Properly splitting up the placeholdered command into an array is not that simple, it would require tokenizing the command (naively splitting by spaces risks changing the semantics) so it's easier to let `/bin/sh` do it for me. however you do have a good point that this will not work on Windows. Instead, I can do the same thing but use `cmd.exe /C`, like in [this StackOverflow post](https://stackoverflow.com/a/13013520). What do you think of that?

Reply:
Yes, I think that would be a good alternative:
```
	invoker := []string{"/bin/sh", "-c"}
	if runtime.GOOS == "windows" {
		invoker = []string{"cmd.exe", "/C"}
	}
	invoker = append(invoker, placeholdered...)
	cmd := exec.Command(invoker[0], invoker[1:]...)
```